test/libsolidity/SMTCheckerTest.cpp

```diff
@@ -47,14 +45,19 @@ SMTCheckerTest::SMTCheckerTest(string const& _filename): SyntaxTest(_filename, E
 	if (!available.cvc4)
 		m_enabledSolvers.cvc4 = false;
 
-	if (m_enabledSolvers.none())
+	auto engine = ModelCheckerEngine::fromString(m_reader.stringSetting("SMTEngine", "all"));
```

Comment:
What does this do on an invalid input? Shouldn't that be reported like above with the exception?

Reply:
Right, added